### PR TITLE
feat(ApplicationCommand): attachment application command option type

### PR DESCRIPTION
**Pull request by amitojsingh366**
**Please describe the changes this PR makes and why it should be merged:**
This pr adds the types for the `ATTACHMENT` application command option type

- [x] update `ApplicationCommandOptionType` enum in both `v8` and `v9` 
- [x] add `ApplicationCommandInteractionDataOptionAttachment` type to both `v8` and `v9` 
- [x] wait for https://github.com/discord/discord-api-docs/pull/4253 to be merged

**Reference Discord API Docs PRs or commits:**
this PR relies on https://github.com/discord/discord-api-docs/pull/4253

> **NOTE:** this is my first time contributing to this project so if anything is wrong please do tell me

### CODE REVIEW

File: deno/payloads/v8/_interactions/_applicationCommands/chatInput.ts [hunk 1]
```diff
@@ -217,6 +219,11 @@ export interface ApplicationCommandInteractionDataOptionNumber
 	focused?: boolean;
 }
 
+export type ApplicationCommandInteractionDataOptionAttachment = InteractionDataOptionBase<
+	ApplicationCommandOptionType.Attachment,
+	APIAttachment
```
Comment (vladfrangu) [inline deno/payloads/v8/_interactions/_applicationCommands/chatInput.ts:224]: Isn't this an id?

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: deno/payloads/v8/_interactions/_applicationCommands/chatInput.ts (hunk 1 above)
Comment (amitojsingh366) [reply deno/payloads/v8/_interactions/_applicationCommands/chatInput.ts:224]: oh shit my bad, yes this is a snowflake id I had kept it as an api attachment when I was still unsure of the structure. 
Fixed it now

**Review by vladfrangu — APPROVED** [review]

File: deno/payloads/v9/_interactions/_applicationCommands/_chatInput/shared.ts [hunk 2]
```diff
@@ -12,6 +12,7 @@ export enum ApplicationCommandOptionType {
 	Role,
 	Mentionable,
 	Number,
+	Attachment,
```
Comment (vladfrangu) [inline deno/payloads/v9/_interactions/_applicationCommands/_chatInput/shared.ts:15]: I think v8 is missing this

**Review by vladfrangu — CHANGES_REQUESTED** [review]
You may want to rerun the build:deno script

File: deno/payloads/v9/_interactions/_applicationCommands/_chatInput/shared.ts (hunk 2 above)
Comment (amitojsingh366) [reply deno/payloads/v9/_interactions/_applicationCommands/_chatInput/shared.ts:15]: added

**Comment by amitojsingh366** [comment]
> You may want to rerun the build:deno script

I get no output:
Screen Shot 2022-01-02 at 10 27 10 PM

**Comment by vladfrangu** [comment]
You're not gonna get output from the script, but you'll get changed files

**Comment by amitojsingh366** [comment]
> You're not gonna get output from the script, but you'll get changed files

Oh ok, my commit was made after the command so I missed what changed. But ig as long as it didn't error it's fine right?

**Review by kyranet — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Comment by Smidul** [comment]
Upstream PR merged 🍕

**Pull request merged**